tools/run_tests/pre_build_csharp.bat

```diff
@@ -38,8 +38,21 @@ cd /d %~dp0\..\..
 set NUGET=C:\nuget\nuget.exe
 
 if exist %NUGET% (
+  @rem Restore Grpc packages by packages since Nuget client 3.4.4 doesnt support restore
```

Comment:
For the vsproject/grpc_csharp_ext.sln, I am not actually sure at the moment why that works but the C# solutions don't. But I got the idea to restore project by project mostly from https://github.com/NuGet/Home/issues/1543